core-client/transports/src/transports/uds.rs

```diff
@@ -0,0 +1,178 @@
+//! JSON-RPC unix domain socket client implementation.
+use crate::{RpcChannel, RpcError};
+use failure::Error;
+use futures::prelude::*;
+use std::{collections::VecDeque, path::Path};
+use tokio::codec::{Framed, LinesCodec};
+use tokio_uds::UnixStream;
+
+/// Connect to a JSON-RPC UDS server.
+pub fn connect<P, T>(path: P) -> impl Future<Item = T, Error = RpcError>
+where
+	P: AsRef<Path>,
+	T: From<RpcChannel>,
+{
+	UnixStream::connect(path)
+		.map(|unix_stream| {
+			let (sink, stream) = Framed::new(unix_stream, LinesCodec::new()).split();
+			let (sink, stream) = UdsClient::new(sink, stream).split();
+			let (rpc_client, sender) = super::duplex(sink, stream);
+			let rpc_client = rpc_client.map_err(|error| eprintln!("{:?}", error));
+			tokio::spawn(rpc_client);
+			sender.into()
+		})
+		.map_err(|error| RpcError::Other(error.into()))
+}
+
+/// Intermediate step in uds pipeline between tokio-uds and duplex sink and
+/// stream converting between io and rpc errors.
+struct UdsClient<TSink, TStream> {
+	sink: TSink,
+	stream: TStream,
+	queue: VecDeque<String>,
+}
+
+impl<TSink, TStream, TError> UdsClient<TSink, TStream>
+where
+	TSink: Sink<SinkItem = String, SinkError = TError>,
+	TStream: Stream<Item = String, Error = TError>,
+	TError: Into<Error>,
+{
+	pub fn new(sink: TSink, stream: TStream) -> Self {
+		Self {
+			sink,
+			stream,
+			queue: VecDeque::new(),
+		}
+	}
+}
+
+impl<TSink, TStream, TError> Sink for UdsClient<TSink, TStream>
```

Comment:
This is essentially the same to the ws `Sink` impl. https://github.com/paritytech/jsonrpc/blob/master/core-client/transports/src/transports/ws.rs#L51. I wonder whether, if this is required for other duplex transports, we should abstract it.